packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1370,113 +1370,94 @@ module.exports = (
       },
       handler: async function accountDestroy(request) {
         log.begin('Account.destroy', request);
-        const form = request.payload;
-        const authPW = form.authPW;
 
-        return customs
-          .check(request, form.email, 'accountDestroy')
-          .then(db.accountRecord.bind(db, form.email))
-          .then(emailRecord => {
-            return totpUtils.hasTotpToken(emailRecord).then(hasToken => {
-              const sessionToken = request.auth && request.auth.credentials;
-
-              // Someone tried to delete an account with TOTP but did not specify a session.
-              // This shouldn't happen in practice, but just in case we throw unverified session.
-              if (!sessionToken && hasToken) {
-                throw error.unverifiedSession();
-              }
+        const { authPW, email: emailAddress } = request.payload;
+        const sessionToken = request.auth && request.auth.credentials;
 
-              // If TOTP is enabled, ensure that the session has the correct assurance level before
-              // deleting account.
-              if (
-                sessionToken &&
-                hasToken &&
-                (sessionToken.tokenVerificationId ||
-                  sessionToken.authenticatorAssuranceLevel <= 1)
-              ) {
-                throw error.unverifiedSession();
-              }
+        await customs.check(request, emailAddress, 'accountDestroy');
```

Comment:
What happens if this throws now that this is outside of the promise chain/try handler? It seems like the behavior will remain the same, the errors were re-thrown previously.

Reply:
All rejected promises returned by route handlers get wrapped by Hapi and then hit this condition in `lib/server.js`, where they go though `AppError.translate` before being serialised to the response:

https://github.com/mozilla/fxa/blob/94fe8d5917ed6c92b87a623936d38e75af4dffc6/packages/fxa-auth-server/lib/server.js#L282-L288

So throwing is fine here.